pkg/network/node/networkpolicy.go

```diff
@@ -736,18 +758,28 @@ func (np *networkPolicyPlugin) handleDeleteNamespace(obj interface{}) {
 	np.updateMatchCache(npns)
 }
 
-func (np *networkPolicyPlugin) refreshNetworkPolicies(refreshFor refreshForType) {
+func (np *networkPolicyPlugin) refreshNamespaceNetworkPolicies() {
 	for _, npns := range np.namespaces {
-		changed := false
 		for _, npp := range npns.policies {
-			if ((refreshFor == refreshForNamespaces) && npp.watchesNamespaces) ||
-				((refreshFor == refreshForPods) && npp.watchesPods) {
-				if np.updateNetworkPolicy(npns, &npp.policy) {
-					changed = true
-				}
+			if npp.watchesNamespaces {
+				npns.mustRecalculate = true
+			}
+		}
+		if npns.mustRecalculate && npns.inUse {
+			np.syncNamespace(npns)
+		}
+	}
+}
+
+func (np *networkPolicyPlugin) refreshPodNetworkPolicies(pod *corev1.Pod) {
+	podNs := np.namespacesByName[pod.Namespace]
+	for _, npns := range np.namespaces {
+		for _, npp := range npns.policies {
+			if (npp.watchesOwnPods && npns == podNs) || npp.watchesAllPods {
+				npns.mustRecalculate = true
 			}
 		}
-		if changed && npns.inUse {
+		if npns.mustRecalculate && npns.inUse {
 			np.syncNamespace(npns)
```

Comment:
Does it make sense to just set `npns.mustSync` on all relevant namespaces, and trigger the runner once? I don't see much advantage to "splitting out" the transaction.